webapp/channels/src/components/get_public_link_modal/get_public_link_modal.test.tsx

```diff
@@ -23,47 +26,45 @@ describe('components/GetPublicLinkModal', () => {
             link: '',
         };
 
-        const wrapper = shallow<GetPublicLinkModal>(
+        const wrapper = shallow(
             <GetPublicLinkModal {...props}/>,
         );
 
         expect(wrapper).toMatchSnapshot();
     });
 
     test('should match snapshot when link is not empty', () => {
-        const wrapper = shallow<GetPublicLinkModal>(
+        const wrapper = shallow(
             <GetPublicLinkModal {...baseProps}/>,
         );
 
         expect(wrapper).toMatchSnapshot();
     });
 
     test('should call getFilePublicLink on GetPublicLinkModal\'s show', () => {
-        const wrapper = shallow<GetPublicLinkModal>(
-            <GetPublicLinkModal {...baseProps}/>,
-        );
+        mountWithIntl(<GetPublicLinkModal {...baseProps}/>);
 
-        wrapper.setState({show: true});
         expect(baseProps.actions.getFilePublicLink).toHaveBeenCalledTimes(1);
         expect(baseProps.actions.getFilePublicLink).toHaveBeenCalledWith(baseProps.fileId);
     });
 
     test('should not call getFilePublicLink on GetLinkModal\'s onHide', () => {
-        const wrapper = shallow<GetPublicLinkModal>(
+        const wrapper = shallow(
             <GetPublicLinkModal {...baseProps}/>,
         );
 
-        wrapper.setState({show: true});
         baseProps.actions.getFilePublicLink.mockClear();
         wrapper.find(GetLinkModal).first().props().onHide();
         expect(baseProps.actions.getFilePublicLink).not.toHaveBeenCalled();
     });
 
     test('should call handleToggle on GetLinkModal\'s onHide', () => {
-        const wrapper = shallow<GetPublicLinkModal>(
-            <GetPublicLinkModal {...baseProps}/>);
+        const wrapper = mountWithIntl(<GetPublicLinkModal {...baseProps}/>);
 
-        wrapper.find(GetLinkModal).first().props().onHide();
-        expect(wrapper.state('show')).toBe(false);
+        act(() => {
+            wrapper.find(GetLinkModal).first().props().onHide();
+        });
+        wrapper.update();
```

Comment:
Maybe (not sure) using shallow there is no need for this `act`. 0/5.

Reply:
Yes, you are right. But due to the limitation of `useEffect`, I shifted the test case to mount which is why `act()` was necessary.